src/common/rsvc.c

```diff
@@ -221,6 +256,13 @@ rsvc_client_complete_rpc(struct rsvc_client *client, const crt_endpoint_t *ep,
 		rsvc_client_process_hint(client, hint, false /* !from_leader */,
 					 ep);
 		return RSVC_CLIENT_RECHOOSE;
+	} else if (rc_svc == -DER_NOTREPLICA) {
+		/* This may happen when a service replica was destroyed. */
+		D_DEBUG(DB_MD, "service not found reply from rank %u: ",
+			ep->ep_rank);
+		rsvc_client_process_error(client, rc_svc, ep);
+		return client->sc_ranks == NULL ?
+			RSVC_CLIENT_PROCEED : RSVC_CLIENT_RECHOOSE;
```

Comment:
Do we also need to update this condition to the following?

    client->sc_ranks->rl_nr == 0